RELEASE/scripts/sl_ascend/sl_cooking.ash

```diff
@@ -1497,3 +1497,375 @@ void consumeStuff()
+
+	if(count(foods) == 0)
+	{
+		print("Couldn't find a way of finishing off our stomach space exactly.", "red");
+		return false;
+	}
+
+	if(!canSimultaneouslyAcquire(foods))
+	{
+		print("Looks like I can't simultaneously acquire all of those items. I'm a bit confused. I'll wait and see if I get unconfused - otherwise, please eat manually.", "red");
+		return false;
+	}
+
+	if(simulate) return true;
+
+	if (count(foods) > 0)
+	{
+		foreach what, howmany in foods
+		{
+			retrieve_item(howmany, what);
+		}
+		if (in_tcrs() && get_property("sl_useWishes").to_boolean() && (0 == have_effect($effect[Got Milk])))
+		{
+			// +15 adv is worth it for daycount
+			// TODO: Some folks have requested a setting to turn this off.
+			makeGenieWish($effect[Got Milk]);
+		}
+		else dealwithMilkOfMagnesium(true);
+
+		foreach what, howmany in foods
+		{
+			slEat(howmany, what);
+		}
+		return true;
+	}
+	return false;
+}
+
+boolean loadDrinks(item[int] item_backmap, float[int] adv, int[int] inebriety)
+{
+	int[item] small_owned;
+	int[item] buyables;
+	int[item] large_owned;
+	int[item] craftables;
+
+	foreach it in $items[]
+	{
+		if ((it.quality == "awesome" || it.quality == "EPIC") && canEat(it) && (it.inebriety > 0) && is_unrestricted(it) && historical_price(it) <= 20000)
+		{
+			int howmany = 1 + inebriety_left()/it.inebriety;
+			if (item_amount(it) > 0 && it.inebriety <= 5)
+			{
+				small_owned[it] = min(item_amount(it), howmany);
+			}
+			if (npc_price(it) > 0)
+			{
+				howmany = min(howmany, my_meat() / npc_price(it));
+				buyables[it] = min(howmany, my_meat() / npc_price(it));
+			}
+			else if (buy_price($coinmaster[hermit], it) > 0)
+			{
+				buyables[it] += min(howmany, my_meat() / 500);
+			}
+			if (item_amount(it) > 0 && it.inebriety > 5)
+			{
+				large_owned[it] = min(item_amount(it), howmany);
+			}
+			if (creatable_amount(it) > 0)
+			{
+				howmany = min(howmany, creatable_amount(it));
+				craftables[it] = howmany;
+			}
+		}
+	}
+
+	void add(item it, boolean crafting, int howmany)
+	{
+		for (int i = 0; i < howmany; i++)
+		{
+			int n = count(inebriety);
+			inebriety[n] = it.inebriety;
+			adv[n] = expectedAdventuresFrom(it);
+			if (crafting)
+			{
+				int turns_to_craft = creatable_turns(it, i + 1, false) - creatable_turns(it, i, false);
+				adv[n] -= turns_to_craft;
+			}
+			item_backmap[n] = it;
+		}
+	}
+
+	foreach it, howmany in small_owned
+	{
+		add(it, false, howmany);
+	}
+	foreach it, howmany in buyables
+	{
+		add(it, false, howmany);
+	}
+	foreach it, howmany in large_owned
+	{
+		add(it, false, howmany);
+	}
+	foreach it, howmany in craftables
+	{
+		add(it, true, howmany);
+	}
+	return true;
+}
+
+item sl_bestNightcap()
+{
+	int[int] inebriety;
+	float[int] adv;
+	item[int] item_backmap;
+
+	loadDrinks(item_backmap, adv, inebriety);
+
+	int best = 0;
+	int n = count(item_backmap);
+	for (int i=1; i < n; i++)
+	{
+		if (adv[i] + inebriety[i] > adv[best] + inebriety[i]) best = i;
```

Comment:
This could probably stand to have a condition of having ode to add inebriety

Reply:
True! Added.